### PR TITLE
task/polygons/base_draw_activity

### PR DESCRIPTION
**Pull request by BarAgam**
1. Action drawing activity
3. BaseActionDrawingFragment
2. BaseMapViewModel 
3. Refactor of ViewerFragment VM to use BaseMapViewModel
4. ExampleActionDrawingFragment with a simple map.

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/view/fragments/BaseDataFragment.java [hunk 1]
```diff
@@ -7,16 +7,13 @@
 import android.view.View;
 import android.view.ViewGroup;
 
-import com.teamagam.gimelgimel.app.GGApplication;
 import com.teamagam.gimelgimel.app.common.base.ViewModels.BaseViewModel;
 
 /**
  * used for VMMV design.
  */
 public abstract class BaseDataFragment<VM extends BaseViewModel>
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/base/view/fragments/BaseDataFragment.java]: rename BaseDataFragment to BaseDataBindingFragment

File: app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/MapModule.java [hunk 2]
```diff
@@ -26,9 +26,14 @@ MainActivity provideMainActivity(Activity activity) {
...
     @Provides
-    @PerActivity
-    ViewerCameraController provideCameraController(MapViewModel mapViewModel) {
-        return mapViewModel;
+    ViewerFragment provideViewerFragment(MainActivity mainActivity) {
+        return mainActivity.getViewerFragment();
     }
 
-}
+
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/MapModule.java]: redundant

File: app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionActivity.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private void initMainFragment() {
+        mMainFragment = getMainFragment();
+        getSupportFragmentManager()
+                .beginTransaction()
+                .add(R.id.draw_action_fragment_container, mMainFragment)
+                .commit();
+    }
+
+    private BaseDrawActionFragment getMainFragment() {
+        return new ExampleDrawActionFragment();
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionActivity.java:35]: If I understood you yesterday, here in future tasks will be a switch case based on the bundle we talked about?
Or we can make this method protected abstract and implement in extending activities...

File: app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionBottomPanelFragment.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.GGApplication;
+import com.teamagam.gimelgimel.app.common.base.view.fragments.BaseFragment;
+
+public class DrawActionBottomPanelFragment extends BaseFragment<GGApplication> {
+
+    private Button mPositiveButton;
+    private Button mCancelButton;
+
+    public DrawActionBottomPanelFragment() {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionBottomPanelFragment.java]: not needed

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+public class BaseMapViewModel<V> extends BaseViewModel<V> {
+
+    private final DisplayMapEntitiesInteractorFactory mMapEntitiesInteractorFactory;
+    private final DisplayVectorLayersInteractorFactory mDisplayVectorLayersInteractorFactory;
+    private final DisplayIntermediateRastersInteractorFactory mDisplayIntermediateRastersInteractorFactory;
+
+    private final GGMapView mGGMapView;
+
+    private DisplayMapEntitiesInteractor mMapEntitiesInteractor;
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java]: m*Display*MapEntitiesInteractor

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    private DisplayVectorLayersInteractor mDisplayVectorLayersInteractor;
+    private DisplayIntermediateRastersInteractor mDisplayIntermediateRastersInteractor;
+
+    protected BaseMapViewModel(
+            DisplayMapEntitiesInteractorFactory mapEntitiesInteractorFactory,
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java]: displayMapEntitiesInteractorFactory

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/ExampleDrawActionViewModel.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import com.teamagam.gimelgimel.app.map.view.GGMapView;
+import com.teamagam.gimelgimel.domain.layers.DisplayVectorLayersInteractorFactory;
+import com.teamagam.gimelgimel.domain.map.DisplayMapEntitiesInteractorFactory;
+import com.teamagam.gimelgimel.domain.rasters.DisplayIntermediateRastersInteractorFactory;
+
+@AutoFactory
+public class ExampleDrawActionViewModel extends BaseMapViewModel<ExampleDrawActionFragment> {
+
+    protected ExampleDrawActionViewModel(
+            @Provided DisplayMapEntitiesInteractorFactory mapEntitiesInteractorFactory,
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/ExampleDrawActionViewModel.java:15]: *display*MapEntitiesInteractorFactory

**Review by teamagam5 — APPROVED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/view/fragments/BaseDataFragment.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/common/base/view/fragments/BaseDataFragment.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/MapModule.java (hunk 2 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/MapModule.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionBottomPanelFragment.java (hunk 4 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionBottomPanelFragment.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionActivity.java (hunk 3 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionActivity.java:35]: Yes, getIntent can be called and it'll return the intent that created this activity.
You can then extract the needed parameter from it.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java (hunk 5 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/BaseMapViewModel.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/ExampleDrawActionViewModel.java (hunk 7 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/ExampleDrawActionViewModel.java:15]: K

**Pull request merged**